sdks/python/apache_beam/io/gcp/gcsio.py

```diff
@@ -87,6 +87,43 @@
 MAX_BATCH_OPERATION_SIZE = 100
 
 
+def proxy_info_from_environment_var(proxy_env_var):
+  """Reads proxy info from the environment and converts to httplib2.ProxyInfo.
+  Args:
+    proxy_env_var: environment variable string to read, http_proxy or
+       https_proxy (in lower case).
+       Example: http://myproxy.domain.com:8080
+  Returns:
+    httplib2.ProxyInfo constructed from the environment string.
+  """
+  proxy_url = os.environ.get(proxy_env_var)
+  if not proxy_url or not proxy_env_var:
+    return httplib2.ProxyInfo(httplib2.socks.PROXY_TYPE_HTTP, None, 0)
```

Comment:
<!--new_thread; commit:4b6f31a090cec94795242c2adec791448acc5f0d; resolved:0-->
Why would `proxy_env_var` ever be `None`?  Also, can we just return `None` here so that we get the same defaults if the env var is empty?

Reply:
I agree proxy_env_var doesn't need to be validated as we are checking proxy_url already.